atc/engine/builder/builder.go

```diff
@@ -21,7 +22,7 @@ const supportedSchema = "exec.v2"
 type StepFactory interface {
 	GetStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.GetDelegate) exec.Step
 	PutStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.PutDelegate) exec.Step
-	TaskStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.TaskDelegate) exec.Step
+	TaskStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.TaskDelegate, atc.UnsafeWorkerOverrides) exec.Step
```

Comment:
Good idea to not pass this along to the resource containers. That'd be a whole can of worms.

Reply:
Makes sense!